ble/Gap.h

```diff
@@ -57,7 +57,7 @@ class Gap {
      * deprecated state to transparenly support existing applications which may
      * have used Gap::ADDR_TYPE_*.
      */
-    enum {
+    enum AddressTypeOverload_t {
```

Comment:
Liyou, could you rename AddressTypeOverload_t into DeprecatedAddressType_t or something like this ? 

I would be very happy if we all start to use the `__deprecated_message` from compiler-polyfill. More information here: https://github.com/ARMmbed/ble/issues/152